### ISSUE TITLE
ContractData: Not updating on change

### ISSUE THREAD

**Issue by RyRy79261** [trufflesuite/drizzle-react-components#13]
```<ContractData contract="ERC223StandardToken" method="balanceOf" methodArgs={[this.props.accounts[0],{from: this.props.accounts[0]}]} />```

```<Faucet contract="ERC223StandardToken" method="faucet" />```

Faucet function signs transaction, updating user balance with 200 tokens. ContractData component displays out of sync icon but does not update to new balance, refreshing page, ContractData displays correct data

**Comment by cNoveron** [comment]
Same issue here!

I made my own React component which takes the color from the first 6 characters of an account, called Blance. There are 3 static Balances I render on load, and below there's this Balance that gets its address string from the state of the input above it.

```javascript
            <input 
              type="number" 
              value={this.state.fromAccount.index} 
              onChange={this.changeIndex}
            />
            <Balance 
              addressStr={this.props.accounts[this.state.fromAccount.index]}
            />
```

![image](https://user-images.githubusercontent.com/29136383/48310099-494fa900-e54d-11e8-87e4-8e7021318aaf.png)

As you can see, the state does change, but ContractData component's props never get updated after rendering for first time.

**Comment by cNoveron** [comment]
@RyRy79261 I think I found the cause of this

[ContractData.js](https://github.com/trufflesuite/drizzle-react-components/blob/master/src/ContractData.js) :trollface: 
```javascript
  constructor(props, context) {
    super(props)

    this.contracts = context.drizzle.contracts

    // Get the contract ABI
    const abi = this.contracts[this.props.contract].abi;

    // Fetch initial value from chain and return cache key for reactive updates.
    var methodArgs = this.props.methodArgs ? this.props.methodArgs : []
    this.dataKey = this.contracts[this.props.contract].methods[this.props.method].cacheCall(...methodArgs)
  }
```
The dataKey is called just once by the constructor. Never to be seen again in render(), except this part.

```javascript
  // If the cache key we received earlier isn't in the store yet; the initial value is still being fetched. 
  if(!(this.dataKey in this.props.contracts[this.props.contract][this.props.method])) {
      return (
        <span>Fetching...</span>
      )
  }
```
Yeah, it talks about an "inital value" implying there should be more values throughout the component's life. But there will be no more updates of dataKey...

**Comment by RyRy79261** [comment]
You biscuit, cheers! Maybe I can get it to work again 🤷‍♂️

**Comment by samajammin** [comment]
@RyRy79261 @carlosescom did either of you find a solution to this? I'm hitting the same issue, where ContractData doesn't react to a state change. Should we be invoking the `cacheCall` somewhere other than the constructor?

**Comment by samajammin** [comment]
Ah, sorry! Should have looked at the https://github.com/trufflesuite/drizzle-react-components/pull/50 PR reference above. Thanks!

**Comment by cNoveron** [comment]
@sbrichards Had any luck making it work? I had to bundle up my own version of drizzle-react-components with Webpack. Just clone my repo if you want, the only caveat I had to deal with was to setup webpack.config to include es-2015 Babel preset.

**Comment by samajammin** [comment]
@carlosescom yes, thanks for checking!

Unfortunately this was not the only issue I came across with this repo, and it appears PR fixes have not been getting merged, so I ended up creating my own ContractData component within my own project & grabbing your `componentDidMount()` code from https://github.com/trufflesuite/drizzle-react-components/pull/50 to get around it. Thanks again 👌

**Comment by honestbonsai** [comment]
Will have to apply these changes to both `master` and `next`.

`master` uses the old context api: PR #5
`next` uses the newer one: PR #50

**Comment by honestbonsai** [comment]
Fixed by #57
